internal/bsr/container.go

```diff
@@ -514,3 +545,40 @@ func (c *container) close(_ context.Context) error {
 
 	return closeError
 }
+
+// Validate checks the signature of the SHA256SUM file,
+// iterates over all files in the SHA256SUM file
+// and verifies that each file checksum is as expected.
+//
+// This function expects that the container's kms keys
+// are loaded into memory and the signature files are
+// verified.
+func (c *container) Validate(ctx context.Context) (BSRChecksumValidation, error) {
+	const op = "bsr.(container).Validate"
+	if len(c.shaSums) == 0 {
+		return nil, fmt.Errorf("%s: missing checksums", op)
+	}
+	if c.keys == nil {
+		return nil, fmt.Errorf("%s: missing keys", op)
+	}
+	checksumValidations := make(BSRChecksumValidation, len(c.shaSums))
+	for fileName, expectedChecksum := range c.shaSums {
+		bsrFileName := BSRFileName(fileName)
+		f, err := c.container.OpenFile(ctx, fileName)
+		if err != nil {
+			checksumValidations[bsrFileName] = BSRChecksumStatus(fmt.Sprintf(bsrFailedCheckTemplate, err.Error()))
+			continue
+		}
+		actualChecksum, err := crypto.Sha256Sum(ctx, f, crypto.WithHexEncoding(true))
+		if err != nil {
+			checksumValidations[bsrFileName] = BSRChecksumStatus(fmt.Sprintf(bsrFailedCheckTemplate, err.Error()))
+			continue
+		}
+		if bytes.Equal(expectedChecksum, actualChecksum) {
+			checksumValidations[bsrFileName] = BSRCheckPass
+			continue
+		}
+		checksumValidations[bsrFileName] = BSRChecksumStatus(fmt.Sprintf(bsrFailedCheckTemplate, "checksum mismatch"))
```

Comment:
I'm wondering if this should be !bytes.Equal with the checksum mismatch error status, and if the fallthrough should be the BSRCheckPass case for readability